db/attachment.go

```diff
@@ -104,23 +104,18 @@ func (db *Database) storeAttachments(doc *Document, newAttachmentsMeta Attachmen
 			if meta["stub"] != true {
 				return nil, base.HTTPErrorf(400, "Missing data of attachment %q", name)
 			}
-
-			revpos, ok := base.ToInt64(meta["revpos"])
-			if !ok || revpos < 1 {
-				return nil, base.HTTPErrorf(400, "Missing/invalid revpos in stub attachment %q", name)
-			}
 			// Try to look up the attachment in ancestor attachments
 			if parentAttachments == nil {
 				parentAttachments = db.retrieveAncestorAttachments(doc, parentRev, docHistory)
 			}
 
+			// Note: in a non-conflict CAS retry, parentAttachments may be nil, because the attachment
+			//  data was persisted prior to the CAS failure writing the doc.  In this scenario the
+			//  incoming doc attachment metadata has already been updated to stub=true to avoid attempting to
+			//  persist the attachment again, even though there is not an attachment on an ancestor.
 			if parentAttachments != nil {
 				if parentAttachment := parentAttachments[name]; parentAttachment != nil {
```

Comment:
If an attachment name is changed in the bucket, this check fails. This causes the version to not be stamped in, and then the attachment gets deleted from the bucket due to the obsolete attachment handling (as it can't find the attachment due to the incorrect version).

Reply:
What do you mean by 'attachment name changed in the bucket'?  What's the scenario where that would happen?

Reply:
Sorry my mistake. Corrected initial comment now.

Reply:
Sounds good.  Have pushed up a tweak to cover that case.